File: runtime/src/main/java/com/siimkinks/sqlitemagic/entity/EntityOperationBuilder.java
package com.siimkinks.sqlitemagic.entity;


import com.siimkinks.sqlitemagic.ConflictAlgorithm;

import androidx.annotation.CheckResult;
import androidx.annotation.NonNull;

interface EntityOperationBuilder<R> extends ConnectionProvidedOperation<R> {
  /**
   * Configure this operation to use provided conflict algorithm.
   *
   * @param conflictAlgorithm One of {@link android.database.sqlite.SQLiteDatabase}
   *                          CONFLICT_* constant values
   * @return Operation builder
   */
  @NonNull
  @CheckResult
  R conflictAlgorithm(@ConflictAlgorithm int conflictAlgorithm);
}

File: runtime/src/main/kotlin/com/siimkinks/sqlitemagic/entity/EntityOperationBuilder.kt
package com.siimkinks.sqlitemagic.entity

import android.database.sqlite.SQLiteDatabase
import androidx.annotation.CheckResult
import com.siimkinks.sqlitemagic.ConflictAlgorithm

interface EntityOperationBuilder<R> : ConnectionProvidedOperation<R> {
  /**
   * Configure this operation to use provided conflict algorithm.
   *
   * @param conflictAlgorithm One of [SQLiteDatabase] CONFLICT_* constant values
   * @return Operation builder
   */
  @CheckResult
  fun conflictAlgorithm(@ConflictAlgorithm conflictAlgorithm: Int): R
}
